src/analysis/special_functions/gaussian.lean

```diff
@@ -334,3 +346,195 @@ begin
+begin
+  -- complete proof using squeeze theorem:
+  rw tendsto_zero_iff_norm_tendsto_zero,
+  refine tendsto_of_tendsto_of_tendsto_of_le_of_le' tendsto_const_nhds _
+    (eventually_of_forall (λ _, norm_nonneg _))
+    ((eventually_ge_at_top (0:ℝ)).mp (eventually_of_forall
+      (λ T hT, vertical_integral_norm_le hb c hT))),
+  rw (by ring : 0 = 2 * |c| * 0),
+  refine (tendsto_exp_at_bot.comp (tendsto_neg_at_top_at_bot.comp _)).const_mul _ ,
+  apply tendsto_at_top_add_const_right,
+  simp_rw [sq, ←mul_assoc, ←sub_mul],
+  refine tendsto.at_top_mul_at_top (tendsto_at_top_add_const_right _ _ _) tendsto_id,
+  exact (tendsto_const_mul_at_top_of_pos hb).mpr tendsto_id,
+end
+
+lemma integrable_cexp_neg_mul_sq_add_real_mul_I (hb : 0 < b.re) (c : ℝ):
+  integrable (λ (x : ℝ), cexp (-b * (x + c * I) ^ 2)) :=
+begin
+  have nm_eq : ∀ (T : ℝ), ‖cexp (-b * (T + c * I) ^ 2)‖ =
+    exp (-(b.re * (T - b.im * c / b.re) ^ 2 - c ^ 2 * (b.im ^ 2 / b.re  + b.re))),
+  { intro T,
+    have : (b.re * T ^ 2 - 2 * b.im * c * T - b.re * c ^ 2) =
+      b.re * (T - b.im * c / b.re) ^ 2 - c ^ 2 * (b.im ^ 2 / b.re  + b.re),
+    { field_simp [hb.ne'], ring },
+    rw [norm_cexp_neg_mul_sq_add_mul_I, this] },
+  refine ⟨(complex.continuous_exp.comp (continuous_const.mul ((continuous_of_real.add
+    continuous_const).pow 2))).ae_strongly_measurable, _⟩,
+  rw ←has_finite_integral_norm_iff,
+  simp_rw [nm_eq, neg_sub _ (c ^ 2 * _),
+    sub_eq_add_neg _ (b.re * _), real.exp_add],
+  suffices : integrable (λ (x : ℝ), exp (-(b.re * x ^ 2))),
+  { exact (integrable.comp_sub_right this (b.im * c / b.re)).has_finite_integral.const_mul _, },
+  simp_rw ←neg_mul,
+  apply integrable_exp_neg_mul_sq hb,
+end
+
+lemma integral_cexp_neg_mul_sq_add_real_mul_I (hb : 0 < b.re) (c : ℝ) :
+  ∫ (x : ℝ), cexp (-b * (x + c * I) ^ 2) = (π / b) ^ (1 / 2 : ℂ) :=
+begin
+  refine tendsto_nhds_unique (interval_integral_tendsto_integral
+    (integrable_cexp_neg_mul_sq_add_real_mul_I hb c) tendsto_neg_at_top_at_bot tendsto_id) _,
+  have C := λ (T : ℝ), integral_boundary_rect_eq_zero_of_differentiable_on
+    (λ z, cexp (-b * z ^ 2)) (-T) (T + c * I)
+    (by { refine differentiable.differentiable_on (differentiable.const_mul _ _).cexp,
+    exact differentiable_pow 2, }),
+  simp only [neg_im, of_real_im, neg_zero, of_real_zero, zero_mul, add_zero, neg_re, of_real_re,
+    add_re, mul_re, I_re, mul_zero, I_im, tsub_zero, add_im, mul_im, mul_one, zero_add,
+    algebra.id.smul_eq_mul, of_real_neg] at C,
+  set I₁ := (λ T, ∫ (x : ℝ) in -T..T, cexp (-b * (x + c * I) ^ 2)) with HI₁,
+  simp only [id.def],
+  simp_rw [←HI₁],
+  let I₂ := λ (T : ℝ), ∫ (x : ℝ) in -T..T, cexp (-b * x ^ 2),
+  let I₄ := λ (T : ℝ), ∫ (y : ℝ) in 0..c, cexp (-b * (T + y * I) ^ 2),
+  let I₅ := λ (T : ℝ), ∫ (y : ℝ) in 0..c, cexp (-b * (-T + y * I) ^ 2),
+  change ∀ (T : ℝ), I₂ T - I₁ T + I * I₄ T - I * I₅ T = 0 at C,
+  have : I₁  = λ (T : ℝ), I₂ T + vertical_integral b c T,
```

Comment:
```suggestion
  refine tendsto_nhds_unique (interval_integral_tendsto_integral
    (integrable_cexp_neg_mul_sq_add_real_mul_I hb c) tendsto_neg_at_top_at_bot tendsto_id) _,
  set I₁ := (λ T, ∫ (x : ℝ) in -T..T, cexp (-b * (x + c * I) ^ 2)) with HI₁,
  let I₂ := λ (T : ℝ), ∫ (x : ℝ) in -T..T, cexp (-b * x ^ 2),
  let I₄ := λ (T : ℝ), ∫ (y : ℝ) in 0..c, cexp (-b * (T + y * I) ^ 2),
  let I₅ := λ (T : ℝ), ∫ (y : ℝ) in 0..c, cexp (-b * (-T + y * I) ^ 2),
  have C : ∀ (T : ℝ), I₂ T - I₁ T + I * I₄ T - I * I₅ T = 0,
  { assume T,
    have := integral_boundary_rect_eq_zero_of_differentiable_on
    (λ z, cexp (-b * z ^ 2)) (-T) (T + c * I)
    (by { refine differentiable.differentiable_on (differentiable.const_mul _ _).cexp,
    exact differentiable_pow 2, }),
    simpa only [neg_im, of_real_im, neg_zero, of_real_zero, zero_mul, add_zero, neg_re, of_real_re,
      add_re, mul_re, I_re, mul_zero, I_im, tsub_zero, add_im, mul_im, mul_one, zero_add,
      algebra.id.smul_eq_mul, of_real_neg] using this },
  simp_rw [id.def, ←HI₁],
  have : I₁ = λ (T : ℝ), I₂ T + vertical_integral b c T,
```
I feel like the proof is more readable if you announce what you are proving beforehand, and it keeps less things around.